### PR TITLE
BAU: Add UsageCriterion for signing to MetadataBackedSignatureValidator

### PR DESCRIPTION
**Pull request by rachelthecodesmith**
Add an additional criterion to ensure only signing certificates are used for signature validation.

Co-Authored-by: Aditya Pahuja <aditya.pahuja@digital.cabinet-office.gov.uk>

### CODE REVIEW

File: saml-security/src/main/java/uk/gov/ida/saml/security/MetadataBackedSignatureValidator.java [hunk 1]
```diff
@@ -40,6 +42,7 @@ private MetadataBackedSignatureValidator(ExplicitKeySignatureTrustEngine explici
         List<Criterion> criteriaSet = new ArrayList<>();
         criteriaSet.add(new EntityIdCriterion(entityId));
         criteriaSet.add(new EntityRoleCriterion(role));
+        criteriaSet.add( new UsageCriterion(UsageType.SIGNING) );
```
Comment (chrisholmes) [inline saml-security/src/main/java/uk/gov/ida/saml/security/MetadataBackedSignatureValidator.java]: This whitespace is inconsistent with the lines around it.

File: saml-security/src/test/java/uk/gov/ida/saml/security/MetadataBackedSignatureValidatorTest.java [hunk 2]
```diff
@@ -126,8 +158,7 @@ private ExplicitKeySignatureTrustEngine getExplicitKeySignatureTrustEngine() thr
         metadataCredentialResolver.setRoleDescriptorResolver(basicRoleDescriptorResolver);
         metadataCredentialResolver.setKeyInfoCredentialResolver(DefaultSecurityConfigurationBootstrap.buildBasicInlineKeyInfoCredentialResolver());
         metadataCredentialResolver.initialize();
-
-        return new ExplicitKeySignatureTrustEngine(metadataCredentialResolver, keyInfoResolver);
+        return metadataCredentialResolver;
```
Comment (adityapahuja) [inline saml-security/src/test/java/uk/gov/ida/saml/security/MetadataBackedSignatureValidatorTest.java]: Could you please undo this change.

Comment (adityapahuja) [inline saml-security/src/test/java/uk/gov/ida/saml/security/MetadataBackedSignatureValidatorTest.java]: Please ignore my previous comment. Github was not showing a wrong function name in diff.

**Comment by chrisholmes** [comment]
LGTM
